CRM/Contribute/BAO/Contribution.php

```diff
@@ -2789,6 +2791,8 @@ public function _gatherMessageValues($input, &$values, $ids = array()) {
     }
     $values['customGroup'] = $customGroup;
 
+    $values['is_pay_later'] = $this->is_pay_later;
```

Comment:
Hi @jitendrapurohit  That would have a different effect: that function is pulling values from the Contribution Page rather than the Contribution itself.  The page may allow pay_later, but the Contribution is not necessarily pay_later.   The comment at https://github.com/civicrm/civicrm-core/blob/master/CRM/Contribute/BAO/Contribution.php#L5034 says `is_pay_later` is one of the values explicitly not to be included there.